build.sh

```diff
@@ -144,133 +146,106 @@ configure() {
+
+    echo "####################################"
+    echo "Building native $MACHINE library"
+    ninja -C build/meson-$MACHINE
 
     if [ "$MACHINE" = "x86_64" ]; then
-        DESTDIR="$PWD/build/release" ninja -C build/meson32 install
+        echo "####################################"
+        echo "Building x86 library"
+        ninja -C build/meson-x86
     fi
 }
 
 package() {
-    LIB="build/release/usr/lib/mangohud/lib64/libMangoHud.so"
-    LIB32="build/release/usr/lib/mangohud/lib32/libMangoHud.so"
-    if [[ ! -f "$LIB" || "$LIB" -ot "build/meson64/src/libMangoHud.so" ]]; then
+    if [[ ! -f "build/meson-$MACHINE/build.ninja" ]]; then
+        configure $@
         build
     fi
+
+    DESTDIR="$PWD/build/release" ninja -C build/meson-$MACHINE install
+
+    if [ "$MACHINE" = "x86_64" ]; then
+        DESTDIR="$PWD/build/release" ninja -C build/meson-x86 install
+    fi
+
     tar --numeric-owner --owner=0 --group=0 \
         -C build/release -cvf "build/MangoHud-package.tar" .
+
+    echo "The 'MangoHud-package.tar' has been created in the build/ folder"
 }
 
 release() {
-    rm build/MangoHud-package.tar
+    [[ -e build/MangoHud-package.tar ]] && rm build/MangoHud-package.tar
     mkdir -p build/MangoHud
-    package
+    package $@
     cp --preserve=mode bin/mangohud-setup.sh build/MangoHud/mangohud-setup.sh
     cp build/MangoHud-package.tar build/MangoHud/MangoHud-package.tar
     tar --numeric-owner --owner=0 --group=0 \
         -C build -czvf build/MangoHud-$VERSION.tar.gz MangoHud
 }
 
 uninstall() {
-    [ "$UID" -eq 0 ] || exec $SU_CMD bash "$0" uninstall
-    rm -rfv "/usr/lib/mangohud"
-    rm -rfv "/usr/share/doc/mangohud"
-    rm -fv "/usr/share/man/man1/mangohud.1"
-    rm -fv "/usr/share/vulkan/implicit_layer.d/mangohud.json"
-    rm -fv "/usr/share/vulkan/implicit_layer.d/MangoHud.json"
-    rm -fv "/usr/share/vulkan/implicit_layer.d/MangoHud.x86.json"
-    rm -fv "/usr/share/vulkan/implicit_layer.d/MangoHud.x86_64.json"
-    rm -fv "/usr/bin/mangohud"
-    rm -fv "/usr/bin/mangohud.x86"
-}
+    echo "#############################"
+    echo "Uninstalling old MangoHud install (if it exists)"
 
-install() {
-    rm -rf "$HOME/.local/share/MangoHud/"
-    rm -f "$HOME/.local/share/vulkan/implicit_layer.d/"{mangohud32.json,mangohud64.json}
-
-    [ "$UID" -eq 0 ] || mkdir -pv "${CONFIG_DIR}"
-    [ "$UID" -eq 0 ] || build
-    [ "$UID" -eq 0 ] || exec $SU_CMD bash "$0" install
-
-    uninstall
-
-    DEFAULTLIB=lib32
-    for i in $DISTRO; do
-        case $i in
-            *arch*)
-            DEFAULTLIB=lib64
-            ;;
-        esac
-    done
-
-    if [ "$MACHINE" != "x86_64" ]; then
-        # Native libs
-        DEFAULTLIB=lib64
+    # The file build/meson-$MACHINE/install-log.txt contains all the files installed by meson
+    if [[ -f $INSTALL_LIST_NATIVE ]]; then
+        $SU_CMD xargs --arg-file=$INSTALL_LIST_NATIVE --verbose rm -rf
```

Comment:
Now that I think of it, this may be dangerous, I think it's better to make the files `$INSTALL_LIST_NATIVE` and `$INSTALL_LIST_MULTILIB` root owned.

Reply:
Wouldn't an attacker just edit build.sh instead then? Or intercept install-log.txt / write additional lines to it?

Reply:
Fair point. If one's machine is running malware as the same user, it's kinda done for. 

Although my suggestion may still be useful for
- not losing the file to `git clean -ffdx` and co ? 
- it kinda belongs to root, conceptually, as the files that got installed are root owned.

🤔

Reply:
Ah yeah, rebuilding over the already made meson build folder causes issues

```
PermissionError: [Errno 13] Permission denied: 'meson-private/../meson-logs/install-log.txt'
```